frontend/coprs_frontend/coprs/logic/builds_logic.py

```diff
@@ -862,6 +865,8 @@ def add(cls, user, pkgs, copr, source_type=None, source_json=None,
             copr_dir=copr_dir,
             bootstrap=bootstrap,
             isolation=isolation,
+            allow_user_ssh=allow_user_ssh,
+            ssh_public_key=ssh_public_key,
```

Comment:
Would it be possible to have this in YAML/JSON format or something...?   We
could have things like:
:
```yaml
---
type: FAS
fas_user: xyzuser
```

```
---
type: plaintext
pub_key: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDj5HauAOur/6+...
```

```
---
type: online
url: https://github.com/praiskup.keys
```

We don't have to implement all the options in the web-UI right now, nor in cli.
But we can be "prepared".

Reply:
This sounds reasonable, not done yet.

Reply:
I thought about this and it is probably not needed now. We IMHO still want one big textarea for user input and then parse the different (types of) keys somewhere in our code. Personally, I would do the parsing on backend (we have workers there) or even on builder. Because we can fetch the keys from FAS or URL right after that.

Right now only plaintext and one key is supported, so the value could only be

```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDj5HauAOur/6+.
```

Seeing some other comments in the PR, we will probably implement support for multiple keys in this PR, so newline would be separator, e.g.

```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDj5HauAOur/6+.
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDj5HauAOur/6+.
```

Once we start to support other methods, the input could be

```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDj5HauAOur/6+.
FAS:xyzuser
https://github.com/praiskup.keys
GitHub:frostyx
```

We can always save the blob to the database as is and parse it when needed. That would be consistent with how e.g. `copr://` external repositories work.

Reply:
> Right now only plaintext and one key is supported

Huh, I've successfully tested before with multiple keys :-)

Reply:
> We IMHO still want one big textarea for user input and then parse the different (types of) keys somewhere in our code.

I'd prefer to have a separate text area in the future, rather than changing the field and start "guessing" the type of input.

My thought was to start doing it right away in the background -- aka giving the user a simple textarea for multi-line key-only input, converting it to the yaml thing in the background, and store into DB.  This way we would have open doors for later enhancements without touching the old entries.  But I agree that this is rather a nice to have thing, there are many such things.